### PR TITLE
feat: add message button to profile using xmtp

### PR DESCRIPTION
**Pull request by elisealix22**
## What does this PR do?

This PR adds a message button placeholder to Profile behind the `messages` feature flag to act as a prototype of sending a `gm` message via XMTP.

Note: currently the button is not styled to spec yet. Also, we are not yet handling the case if the person on the receiving end of the message isn't on XMTP. Those will be added as follow-ups!

<details>
<summary>Screenshots</summary>

| With `messages` FF | Without `messages` FF |
|--|--|
| Screen Shot 2022-10-03 at 3 25 20 PM  |  Screen Shot 2022-10-03 at 3 24 58 PM|
</details>
## How should this be tested?

- Sign in on testnet (or allowlist your profile in the `messages` feature flag)
- Navigate to someone else's profile (elisetest.test is on the XMTP network)
- Tap message
- You should be prompted to sign in with XMTP
- A `gm` message should be sent to the receiving profile as long as they are on XMTP

### CODE REVIEW

File: src/components/Profile/Details.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import Follow from '@components/Shared/Follow';
 import Markup from '@components/Shared/Markup';
+import Message from '@components/Shared/Message';
```
Comment (bigint) [inline src/components/Profile/Details.tsx:3]: **suggestion:** If we are not using the `Message` in multiple places it is better to add it inside `src/components/Profile/`

File: src/components/Shared/Message.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  const currentProfile = useAppStore((state) => state.currentProfile);
+  const { data: signer, isError, isLoading } = useSigner();
+
+  async function sendGm(): Promise<void> {
+    if (!currentProfile || isError || !signer) {
+      toast.error(SIGN_WALLET);
+      return;
+    }
+
+    //TODO(elise): Swap this out with opening the inbox. This is just a prototype.
```
Comment (bigint) [inline src/components/Shared/Message.tsx:24]:
```suggestion
    // TODO(elise): Swap this out with opening the inbox. This is just a prototype.
```

File: src/components/Shared/Message.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+interface Props {
+  profile: Profile;
+}
+
+const Message: FC<Props> = ({ profile }) => {
+  const currentProfile = useAppStore((state) => state.currentProfile);
+  const { data: signer, isError, isLoading } = useSigner();
+
+  async function sendGm(): Promise<void> {
```
Comment (bigint) [inline src/components/Shared/Message.tsx:18]: **suggestion:** can we make it as arrow function 🙏🏼

**Review by bigint — CHANGES_REQUESTED** [review]
@elisealix22 left some comments, can we also resolve the conflicts 🙏🏼

**Comment by bigint** [comment]
@elisealix22 We renamed `getFeatureEnabled` to `isFeatureEnabled` 🙇🏼

File: src/components/Profile/Details.tsx (hunk 1 above)
Comment (elisealix22) [reply src/components/Profile/Details.tsx]: love it! moved it over to profile.

File: src/components/Shared/Message.tsx (hunk 3 above)
Comment (elisealix22) [reply src/components/Shared/Message.tsx]: done!

**Comment by elisealix22** [comment]
@bigint Should be all set now! Thank you so much for the quick feedback 🙇‍♀️

**Comment by bigint** [comment]
Screenshot 2022-10-05 at 11 12 02 PM

I just swapped text with icon for saving some space 🙇🏼

**Pull request merged**